Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -273,12 +257,31 @@ private bool TryDeserializeField(FieldInfo field, IIonReader reader, IonType ion
             // for some Ion types need to be consumed in order to advance the reader.
             deserialized = ionSerializer.Deserialize(reader, field.FieldType, ionType);
             
-            if (options.IgnoreReadOnlyFields && field.IsInitOnly)
+            return !IgnoreDeserializedField(field, deserialized);
+        }
+
+        private bool IgnoreDeserializedProperty(object deserialized)
+        {
+            return options.IgnoreDefaults && deserialized == default;
+        }
+        
+        private bool IgnoreDeserializedField(FieldInfo field, object deserialized)
+        {
+            // Check if this field is really a backing field for a readonly property and
+            // if so, apply the ignore property logic instead of the ignore field logic.
+            if (!options.IgnoreReadOnlyProperties && IsBackingFieldForReadonlyProperty(field))
             {
-                return false;
+                return IgnoreDeserializedProperty(deserialized);
             }
+            
+            return (options.IgnoreReadOnlyFields && field.IsInitOnly) || 
+                   (options.IgnoreDefaults && deserialized == default);
+        }
 
-            return !options.IgnoreDefaults || deserialized != default;
+        private bool IsBackingFieldForReadonlyProperty(FieldInfo field)
+        {
+            var readonlyProperties = this.GetValidProperties(true).Where(IsReadOnlyProperty);
+            return readonlyProperties.Any(p => field.Name == $"<{p.Name}>k__BackingField");
```

Comment:
Is string matching the only way to do this?

Reply:
This name comparison is the only way I know of to determine that a given field is a backing field for a given property.